compiler/rustc_feature/src/removed.rs

```diff
@@ -1,17 +1,20 @@
 //! List of the removed feature gates.
 
+use std::num::NonZero;
+
 use rustc_span::sym;
 
 use super::{Feature, to_nonzero};
 
 pub struct RemovedFeature {
     pub feature: Feature,
     pub reason: Option<&'static str>,
+    pub pull: Option<NonZero<u32>>,
 }
 
 macro_rules! declare_features {
     ($(
-        $(#[doc = $doc:tt])* (removed, $feature:ident, $ver:expr, $issue:expr, $reason:expr),
+        $(#[doc = $doc:tt])* (removed, $feature:ident, $ver:expr, $issue:expr, $reason:expr, $pull:expr),
```

Comment:
can you avoid changing each call site to pass in a None? that will make `git blame` harder to run when we add this pull request info in the future.

you could do that by making the argument optional and default to none:
```suggestion
        $(#[doc = $doc:tt])* (removed, $feature:ident, $ver:expr, $issue:expr, $reason:expr $(, $pull:literal)? ),
```
and then the call sites will either be the same as before, or `(removed, abi_amdgpu_kernel, "1.77.0", Some(51575), None, 141642)`; and below you would have `pull: None $( .or(to_nonzero(Some($pull))) )?`.

make sure to squash your commits after so the `None` argument you added doesn't show up in the git history.